qa/rpc-tests/test_framework/blocktools.py

```diff
@@ -58,20 +64,26 @@ def serialize_script_num(value):
         r.append(0x80 if neg else 0)
     elif neg:
         r[-1] |= 0x80
+    r.insert(0, len(r))
     return r
 
 # Create a coinbase transaction, assuming no miner fees.
 # If pubkey is passed in, the coinbase output will be a P2PK output;
 # otherwise an anyone-can-spend output.
-def create_coinbase(height, pubkey = None):
+def create_coinbase(height, pubkey = None, amount = 0):
     coinbase = CTransaction()
-    coinbase.vin.append(CTxIn(COutPoint(0, 0xffffffff), 
-                ser_string(serialize_script_num(height)), 0xffffffff))
+    # coinbase transaction scriptsigs must be at least 2 bytes
```

Comment:
this comment refers to [this](https://github.com/ElementsProject/elements/blob/elements-0.14.1/src/validation.cpp#L534) condition. Come to think of it, perhaps that check is incorrect. It seems to assume that the height at the start of the script must use a two-byte encoding (which is only true for values > 16)

Reply:
is this upstreamable?

Reply:
you mean the comment or the patch to make the scriptsig at least 2 bytes? I have a feeling that this requirement of the scriptsig being at least 2 bytes is a mistake. But I can't say I understand the motivation of it well enough to be confident. I imagine it's not a problem in the wild, as the block height most likely will get encoded as at least two bytes in the script. But in tests where the height < 16, it is.